pkg/operators/ebpf/stats.go

```diff
@@ -0,0 +1,535 @@
+		return fmt.Errorf("enabling bpf stats: %w", err)
+	}
+
+	return nil
+}
+
+func (i *ebpfOperator) InstantiateDataOperator(
+	gadgetCtx operators.GadgetContext, paramValues api.ParamValues,
+) (operators.DataOperatorInstance, error) {
+	cfg, ok := gadgetCtx.GetVar("config")
+	if !ok {
+		return nil, fmt.Errorf("missing configuration")
+	}
+	v, ok := cfg.(*viper.Viper)
+	if !ok {
+		return nil, fmt.Errorf("invalid configuration format")
+	}
+
+	emitStatsAnn := v.GetBool(fmt.Sprintf("annotations.operator.%s.%s", i.Name(), EmitStatsAnn))
+	if !emitStatsAnn {
+		return nil, nil
+	}
+
+	if i.gadgetObjs == nil {
+		return nil, fmt.Errorf("bpfstats aren't enabled")
+	}
+
+	var err error
+
+	instance := &ebpfOperatorDataInstance{
+		bpfOperator:  i,
+		done:         make(chan struct{}),
+		programStats: paramValues[ParamProgramStats] == "true",
+	}
+
+	instance.ds, err = gadgetCtx.RegisterDataSource(datasource.TypeArray, StatsDSName)
+	if err != nil {
+		return nil, err
+	}
+
+	intervalAnn := paramValues[ParamGadgetStatisticsInterval]
+	if intervalAnn == "" {
+		intervalAnn = "1000ms"
+	}
+	instance.interval, err = time.ParseDuration(intervalAnn)
+	if err != nil {
+		return nil, fmt.Errorf("parsing duration: %w", err)
+	}
+
+	countAnn := paramValues[ParamGadgetStatisticsCount]
+	if countAnn == "" {
+		countAnn = "0"
+	}
+	instance.count, err = strconv.Atoi(countAnn)
+	if err != nil {
+		return nil, fmt.Errorf("parsing count: %w", err)
+	}
+
+	instance.ds.AddAnnotation(api.FetchIntervalAnnotation, intervalAnn)
+	instance.ds.AddAnnotation(api.FetchCountAnnotation, countAnn)
+	instance.ds.AddAnnotation("cli.clear-screen-before", "true")
+
+	if nodeName, ok := os.LookupEnv("NODE_NAME"); ok {
+		instance.nodeNameField, err = instance.ds.AddField("nodeName", api.Kind_String,
+			datasource.WithAnnotations(map[string]string{
+				metadatav1.ColumnsWidthAnnotation: "16",
+				"metrics.type":                    "key",
+			}),
+		)
+		if err != nil {
+			return nil, err
+		}
+		instance.nodeName = nodeName
+	}
+
+	// gadget-specific fields
+	instance.gadgetIDField, err = instance.ds.AddField("gadgetID", api.Kind_String,
+		datasource.WithAnnotations(map[string]string{
+			metadatav1.ColumnsWidthAnnotation: "8",
+			// NOT REALLY SURE
+			//"metrics.type": "key",
+		}),
+	)
+	if err != nil {
+		return nil, err
+	}
+	instance.gadgetNameField, err = instance.ds.AddField("gadgetName", api.Kind_String,
+		datasource.WithAnnotations(map[string]string{
+			metadatav1.ColumnsWidthAnnotation: "16",
+			"metrics.type":                    "key",
+		}),
+	)
+	if err != nil {
+		return nil, err
+	}
+	instance.gadgetImageField, err = instance.ds.AddField("gadgetImage", api.Kind_String,
+		datasource.WithAnnotations(map[string]string{
+			metadatav1.ColumnsWidthAnnotation: "16",
+			"metrics.type":                    "key",
+		}),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// low-level fields
+	// TODO: how to disable it based on the programs param?
```

Comment:
IIRC, we instantiate twice the operator: one for the `GetGadgetInfo` and another for the `RunGadget`. For `GetGadgetInfo` programStats will be always false, but for the `RunGadget`, it should carry the proper value. Mmmm why can't we use `programStats`? I'm sure I'm missing something.

Reply:
The first call doesn't get any parameters and it's the result of this first call which is used by the CLI operator to set the default columns, so I haven't found a way to change the list of columns (or their visibility) based on a param.

Reply:
What about a [custom param](https://github.com/inspektor-gadget/inspektor-gadget/pull/3485) that adds an annotation to the `emitstats`?

Reply:
I'd like to avoid this discussion for now. Once we agree on the whole idea, we can merge it and I can try to improve it later on.

Reply:
Ref #4318